app/scripts/services/DataRepository.js

```diff
@@ -204,26 +246,31 @@ console.log('Error...', error);
    * @param callback {Function|null}
    */
   this.persistBrand = function persistBrand(brand, successMsg, callback) {
-    if (!brand.hasId()) {
-console.log('Brand does not have an id');
-      // This is an insert.
+    if (brand.hasId()) {
+
+      // This is an update.
+      // Determine the index of the record in the array from the model's id.
+      var brandIndex = this.determineIndexFromBrandModel(brand);
+      // Create a reference to the object in the array.
+      var brandSource = this.recordSets.brands[brandIndex];
+      // Overwrite the record with the values from the model.
+      this.populateRecordFromBrandModel(brandSource, brand);
+      // Save the record back to the store.
+      this.update('brands', brandIndex, successMsg, callback);
```

Comment:
These comments are exceptionally helpful in keeping how firebase functions straight 👌. Thank you for them!!

Reply:
Ah, thank you. Yeah, I put the comments in because when I looked at the code it looked really complicated and it wasn't clear at all what was going on, so I essentially commented every line!